src/index.js

```diff
@@ -305,19 +339,22 @@ function renderToString(vnode, context, opts, inner, isSvgMode, selectValue) {
 
 			if (child != null && child !== false) {
 				let childSvgMode =
-						nodeName === 'svg'
-							? true
-							: nodeName === 'foreignObject'
-							? false
-							: isSvgMode,
-					ret = renderToString(
-						child,
-						context,
-						opts,
-						true,
-						childSvgMode,
-						selectValue
-					);
+					nodeName === 'svg'
+						? true
+						: nodeName === 'foreignObject'
+						? false
+						: isSvgMode;
+
+				// TODO: we can't rely on this return-value...
+				let ret = renderToString(
```

Comment:
Not quite sure how to handle this, we could try utilizing a custom array as last argument that scopes the result to this child but then again we merge adjacent nodes from lastpass, ...

Reply:
The only solution I can come up with is to track the iteration via a simple stack and push some sort of marker to that for when a node should be closed. Those can be stored in a second arrays and popped whenever a marker is encountered in the "work" stack.

```js
let output = []
let stack = [vnode]
let closing = [];
let item;
while ((item = stack.pop()) !== undefined) {
  // If we encounter a marker we need to close the element
  if (item === null) {
    output.push(closing.pop())
    continue;
  }

  output.push(`<${item.name}>`, null); // <-- push marker
  closing.push(`</${item.name}>`);

  stack.push(...item.children);
}
```